### PR TITLE
create POST /api/articles/:article_id/comments endpoint

### PR DESCRIPTION
**Pull request by ViktoriiaPavliukh**

### CODE REVIEW

File: __tests__/api.test.js [hunk 1]
```diff
@@ -169,4 +169,50 @@ describe("app", () => {
...
+      .expect(201)
+      .then((response) => {
+        expect(response.body.comment).toMatchObject({
+          article_id: 7,
+          body: "This is a new comment.",
+          author: "icellusedkars"
+        });
+      });
+    });
+  });
```
Comment (Augs0) [inline __tests__/api.test.js]: We should also have another 201 test that ignores extra properties as long as username and body are included e.g. 
```
{
      username: "icellusedkars",
      body: "This is a new comment.",
      randomKey: "ignore me"
    };
```

File: __tests__/api.test.js [hunk 2]
```diff
@@ -169,4 +169,50 @@ describe("app", () => {
...
+    return request(app)
+      .post("/api/articles/1000/comments")
+      .send(newComment)
+      .expect(404)
+      .then((response) => {
+      expect(response.body.msg).toBe(
+        "No article found for article_id: 1000"
+      );
+    });
+  });
```
Comment (Augs0) [inline __tests__/api.test.js:263]: We also need a 400 for an invalid id e.g. /articles/not-a-number/comments

Comment (Augs0) [reply __tests__/api.test.js:263]: Finally, we also need a 404 test for a non-existent username

File: app.js [hunk 3]
```diff
@@ -35,8 +45,8 @@ app.use((err, req, res, next) => {
   }
 });
 
-app.use((err, req, res, next) => {
-  res.status(500).send({ msg: "Internal Server Error"});
-})
+// app.use((err, req, res, next) => {
```
Comment (Augs0) [inline app.js]: Please remove unused code

File: controllers/article.controllers.js [hunk 4]
```diff
@@ -43,3 +44,22 @@ exports.getCommentsByArticleId = (req, res, next) => {
       next(err);
     });
 };
+
+exports.postComment = (req, res, next) => {
+  const { article_id } = req.params;
+  const { username, body } = req.body;
+
+  selectArticleById(article_id)
```
Comment (Augs0) [inline controllers/article.controllers.js]: This is not quite right. Remember, you have access to the article_id from your params. We should pass this to the model and access the right article in our SQL statement

File: models/article.models.js [hunk 5]
```diff
@@ -39,6 +39,19 @@ exports.selectCommentsByArticleId = (article_id) => {
     })
 };
 
+exports.addComment = (article_id, username, body) => {
+  return db
+    .query(
+      "INSERT INTO comments (article_id, author, body) VALUES ($1, $2, $3) RETURNING *;",
```
Comment (Augs0) [inline models/article.models.js:45]: It seems you are already using the article_id as intended here so the select in your controller is not needed. Remember, some errors should be handled in here...

**Review by Augs0 — CHANGES_REQUESTED** [review]
Almost there Viktoriia, well done. I'd like you to add some further testing and to fix your controller. Please resubmit once you have made the requested changes.

File: app.js [hunk 6]
```diff
@@ -18,15 +21,26 @@ app.get("/api/articles/:article_id", getArticleById);
...
+app.post("/api/articles/:article_id/comments", postComment);
+
 
 app.use((err, req, res, next) => {
   if (err.code === "22P02") {
     res.status(400).send({ msg: "Invalid id" });
-  } else {
+  } else if (err.code === "23502") {
+    res
+      .status(400)
```
Comment (dappernerddesigns) [inline app.js:32]: Your status code and error message don't match here.

File: controllers/article.controllers.js [hunk 7]
```diff
@@ -43,3 +45,24 @@ exports.getCommentsByArticleId = (req, res, next) => {
...
+    .then(() => {
+      addComment(article_id, username, body)
+        .then((comment) => {
+          res.status(201).send({ comment: comment });
+        })
+        .catch((err) => {
+          next(err);
+        });
+    })
+    .catch((err) => {
```
Comment (dappernerddesigns) [inline controllers/article.controllers.js:56]: Why are there two catch blocks here?

**Review by dappernerddesigns — CHANGES_REQUESTED** [review]
Looks like you hit most of August's points.

A few areas I'd like you to work on before you move on. You have an error handler that is sending back a 400 status and a message with a 404, its a little mis leading.

In your article controller you have two functions that have double catch blocks on them. That shouldn't be there, you've nested your functionality. In order to make your checkExists work you may need a promise all. Never nest promises 😄 If you could address both of those controller functions and resubmit please.

**Review by nc-poppy-m — APPROVED** [review]
Good to merge!

**Pull request merged**